src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { Component, ReactNode } from 'react'
+import BootstrapModal from 'react-bootstrap/Modal'
+import { Button } from '../Button'
+import { ButtonColor } from '../Button/interfaces'
+import { ButtonsAlignment } from './interfaces'
+
+interface Props {
+  /** Defines if the modal should be visible.*/
+  show: boolean
+  /** Toggles the modal visibility.*/
+  toggle(): void
+  /** Defines the title of the modal.*/
+  title?: string
+  /** Defines the body of the modal.*/
+  body?: ReactNode
+  /**
+   * Renders a close button in modal header.
+   * @default true
+   **/
+  showHeaderCloseButton?: boolean
+  /**
+   * Defines if the modal should be vertically centered.
+   * @default false
+   **/
+  verticallyCentered?: boolean
+  /**
+   * Defines the buttons alignment.
+   * @default "edges"
+   **/
+  buttonsAlignment?: ButtonsAlignment
+  /**
+   * Optional close button properties.
+   **/
+  closeButton?: {
+    /**
+     * Optional label for the close button.
+     * @default "Close"
+     **/
+    label?: string
+    /**
+     * Optional color for the close button.
+     * @default "secondary"
+     **/
+    color?: ButtonColor
+    /**
+     * Optional callback for the close button.
+     **/
+    callback?(): void
+  }
+  /**
+   * Optional middle button properties.
+   **/
+  middleButton?: {
+    /**
+     * Optional label for the middle button.
+     * @default "Another action"
+     **/
+    label?: string
+    /**
+     * Optional color for the middle button.
+     * @default "warning"
+     **/
+    color?: ButtonColor
+    /**
+     * Optional callback for the middle button.
+     **/
+    callback?(): void
+  }
+  /**
+   * Optional success button properties.
+   **/
+  successButton?: {
+    /**
+     * Optional label for the success button.
+     * @default "Confirm"
+     **/
+    label?: string
+    /**
+     * Optional color for the success button.
+     * @default "primary"
+     **/
+    color?: ButtonColor
+    /**
+     * Optional callback for the success button.
+     **/
+    callback?(): void
+  }
+}
+
+/**
+ * Add dialogs for lightboxes, user notifications, or completely custom content.
+ */
+
+class Modal extends Component<Props, {}> {
```

Comment:
This can be rewritten as a functional component

Reply:
I agree.